dot/network/message.go

```diff
@@ -83,6 +85,36 @@ func decodeMessage(r io.Reader) (m Message, err error) {
 	case ConsensusMsgType:
 		m = new(ConsensusMessage)
 		err = m.Decode(r)
+	case RemoteCallRequestType:
+		m = new(RemoteCallRequest)
+		err = m.Decode(r)
+	case RemoteCallResponseType:
+		m = new(RemoteCallResponse)
+		err = m.Decode(r)
+	case RemoteChangesRequestType:
+		m = new(RemoteChangesRequest)
+		err = m.Decode(r)
+	case RemoteChangesResponseType:
+		m = new(RemoteChangesResponse)
+		err = m.Decode(r)
+	case RemoteReadRequestType:
+		m = new(RemoteReadRequest)
+		err = m.Decode(r)
+	case RemoteReadResponseType:
+		m = new(RemoteReadResponse)
+		err = m.Decode(r)
+	case RemoteReadChildRequestType:
+		m = new(RemoteReadChildRequest)
+		err = m.Decode(r)
+	case RemoteReadChildResponseType:
+		m = new(RemoteReadResponse)
+		err = m.Decode(r)
+	case RemoteHeaderRequestType:
+		m = new(RemoteHeaderRequest)
+		err = m.Decode(r)
+	case RemoteHeaderResponseType:
+		m = new(RemoteHeaderResponse)
+		err = m.Decode(r)
```

Comment:
could you add a different decode function to deal with these? the message IDs will be as per above, and whether it's a request or response will be based on context (ie. if you send the request, you expect a response; and if the stream is opened, you would expect a request)